apigw-lambda-rekognition/main.tf

```diff
@@ -0,0 +1,145 @@
+variable "region" {}
+
+provider "aws" {
+  region  = "${var.region}"
+}
+variable "prefix" {
+  description = "Prefix for S3 bucket name"
+  type        = string
+}
+resource "aws_s3_bucket" "upload_bucket" {
+  bucket = "${lower(var.prefix)}-s3-upload"
+}
+resource "aws_sns_topic" "sns_topic" {
+  name = "SNS-Topic"
+}
+resource "aws_iam_role" "lambda_role" {
+  name = "${lower(var.prefix)}-lambda_execution_role"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Principal = {
+          Service = "lambda.amazonaws.com"
+        },
+        Action = "sts:AssumeRole"
+      }
+    ]
+  })
+}
+resource "aws_iam_role_policy" "lambda_policy" {
+  name = "lambda_policy"
+  role = aws_iam_role.lambda_role.id
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Action = [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ],
+        Resource = "*"
+      },
+      {
+        Effect = "Allow",
+        Action = [
+          "s3:PutObject",
+          "s3:GetObject",
+          "s3:ListBucket"
+        ],
+        Resource = [
+          aws_s3_bucket.upload_bucket.arn,
+          "${aws_s3_bucket.upload_bucket.arn}/*"
+        ]
+      },
+      {
+        Effect = "Allow",
+        Action = "sns:Publish",
+        Resource = aws_sns_topic.sns_topic.arn
+      },
+      {
+        Effect = "Allow",
+        Action = "rekognition:DetectModerationLabels",
+        Resource = "*"
+      }
+    ]
+  })
+}
+resource "aws_lambda_function" "generate_presigned_url" {
```

Comment:
suggest explicitly creating the CW log group so that it will be deleted with the stack.